### PR TITLE
feat: create z-index CSS variables for onyx

### LINKED ISSUE #108: Define variables for z-index
Define which component layers we will need inside the application and which z-index they should have.
Create corresponding CSS variables `onyx-elevation-<name>` for the z-index.

Possible reference: https://atlassian.design/foundations/elevation

### PR DESCRIPTION
**Pull request by BoppLi**
relates to #108 

Creates a basic concept for the usage of z-indices in **onyx**. Verifies combinations of layered components via a POC page inside the alpha demo app.
For testing, run the alpha demo app and open http://localhost:5173/layout-demo .

The defined variables are
```css
:root {
  --onyx-z-index-overlay: 60;
  --onyx-z-index-top-nav: 50;
  --onyx-z-index-page-loader: 40;
  --onyx-z-index-global-info: 30;
  --onyx-z-index-flyout: 20;
  --onyx-z-index-tooltip: 10;
}
```

Follow-up ticket for this is [#374](https://github.com/SchwarzIT/onyx/issues/374)

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] All changes are documented in the documentation app (folder `apps/docs`)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: apps/alpha-test-app/src/router/index.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  {
+    path: "/",
+    name: "home",
+    component: HomeView,
+  },
+  {
+    path: "/layout-demo",
+    name: "layoutDemo",
+    component: () => import("../views/LayoutDemoView.vue"),
+    meta: { requiresAuth: false },
```
Comment (larsrickert) [inline apps/alpha-test-app/src/router/index.ts:17]:
```suggestion
    meta: { requiresAuth: false },
```

We don't have our Vue Blueprint here so this is not supported^^

File: apps/alpha-test-app/src/router/index.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import type { RouteRecordRaw } from "vue-router";
+import HomeView from "../views/HomeView.vue";
+
+/** List of all routes the user can navigate to.
+ *  Please declare, whether authentication is enabled/disabled via the meta parameter.
```
Comment (larsrickert) [inline apps/alpha-test-app/src/router/index.ts:5]: We don't have authentication, right?

File: packages/sit-onyx/src/styles/z-indices.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// all z-indices used in the onyx design system
+:root {
+  --onyx-z-index-overlay: 60;
+  --onyx-z-index-top-nav: 50;
+  --onyx-z-index-page-loader: 40;
+  --onyx-z-index-global-info: 30;
+  --onyx-z-index-flyout: 20;
+  --onyx-z-index-tooltip: 10;
```
Comment (larsrickert) [inline packages/sit-onyx/src/styles/z-indices.scss:8]:
```suggestion
  --onyx-z-index-overlay: 60;
  --onyx-z-index-navigation: 50;
  --onyx-z-index-page-overlay: 40;
  --onyx-z-index-page-raised: 30;
  --onyx-z-index-flyout: 20;
  --onyx-z-index-popover: 10;
```

I would like to decouple the z-index from specific components. Do we need a z-index for flyout and tooltip? If we implement them with the Popover/Dialog API they will be placed in the [CSS top layer](https://developer.mozilla.org/en-US/docs/Glossary/Top_layer)

File: apps/alpha-test-app/src/views/HomeView.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+</script>
+
+<template>
+  <div class="page">
+    <div>
+      <p>{{ t("message") }} in {{ locale }}</p>
+      <button @click="locale = 'de-DE'">Deutsch</button>
+      <button @click="locale = 'en-US'">English</button>
+    </div>
+    <button @click="router.push('/layout-demo')">Move to layout demo</button>
```
Comment (larsrickert) [inline apps/alpha-test-app/src/views/HomeView.vue:34]: We can already use our OnyxButton :)

File: apps/alpha-test-app/src/views/LayoutDemoView.vue
Comment (larsrickert) [inline apps/alpha-test-app/src/views/LayoutDemoView.vue:1]: To be honest I don't really understand this page / its hard for me to navigate it 😅

File: apps/alpha-test-app/src/components/molecules/LayoutDemoOptionsMolecule.vue [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  get: () => props.modelValue,
+  set: (value) => emit("update:modelValue", value),
+});
+</script>
+
+<template>
+  <div class="options">
+    Demo options
+    <template v-for="(value, label) of options" :key="label">
+      <label v-if="typeof value === 'boolean'">
```
Comment (larsrickert) [inline apps/alpha-test-app/src/components/molecules/LayoutDemoOptionsMolecule.vue:25]: We can use our OnyxCheckbox :)

File: apps/alpha-test-app/src/views/LayoutDemoView.vue
Comment (larsrickert) [reply apps/alpha-test-app/src/views/LayoutDemoView.vue:1]: If we want to keep this, wouldn't it be better suited in Storybook of sit-onyx like the GridPlayground?
Something like a "LayerPlayground"  / "ZIndexPlayground"

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: apps/alpha-test-app/src/router/index.ts (hunk 1 above)
Comment (BoppLi) [reply apps/alpha-test-app/src/router/index.ts:17]: 😅

Comment (BoppLi) [reply apps/alpha-test-app/src/router/index.ts:17]:
```suggestion
```

File: apps/alpha-test-app/src/components/molecules/LayoutDemoOptionsMolecule.vue (hunk 5 above)
Comment (BoppLi) [reply apps/alpha-test-app/src/components/molecules/LayoutDemoOptionsMolecule.vue:25]: is it ok if I do it in the[ follow-up ticket](https://github.com/SchwarzIT/onyx/issues/374)? I wanted to make a very basic initial draft.
I'll add it to the AC

File: apps/alpha-test-app/src/views/LayoutDemoView.vue
Comment (BoppLi) [reply apps/alpha-test-app/src/views/LayoutDemoView.vue:1]: see [follow up ticket](https://github.com/SchwarzIT/onyx/issues/374)

This is the initial POC to try out collisions of different layer components as base of discussion

File: packages/sit-onyx/src/styles/z-indices.scss (hunk 3 above)
Comment (BoppLi) [reply packages/sit-onyx/src/styles/z-indices.scss:8]: Please see https://github.com/SchwarzIT/onyx/issues/108#issuecomment-1952587571

Comment (BoppLi) [reply packages/sit-onyx/src/styles/z-indices.scss:8]: renaming will be discussed with Jannick, he was not available  :(

File: apps/alpha-test-app/src/router/index.ts (hunk 2 above)
Comment (BoppLi) [reply apps/alpha-test-app/src/router/index.ts:5]: sorry, yep. will clean that up

File: packages/sit-onyx/src/styles/z-indices.scss [hunk 6]
```diff
@@ -0,0 +1,10 @@
+// all z-indices that are needed in the onyx design system
+:root {
+  --onyx-z-index-overlay: 70;
+  --onyx-z-index-nav-bar: 60;
```
Comment (larsrickert) [inline packages/sit-onyx/src/styles/z-indices.scss:4]:
```suggestion
  --onyx-z-index-navigation: 60;
```

This layer would be used for the header and sidebar, right? So its not only for the nav bar

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/styles/z-indices.scss [hunk 7]
```diff
@@ -0,0 +1,10 @@
+// all z-indices that are needed in the onyx design system
+:root {
+  --onyx-z-index-overlay: 70;
+  --onyx-z-index-navigation: 60;
+  --onyx-z-index-content-overlay: 50;
+  --onyx-z-index-notification: 40;
+  --onyx-z-index-sticky-content: 30;
+  --onyx-z-index-flyout: 20;
+  --onyx-z-index-tooltip: 10;
```
Comment (BoppLi) [inline packages/sit-onyx/src/styles/z-indices.scss:9]: merge them
```suggestion
  --onyx-z-index-flyout: 10;
```

**Review by larsrickert — APPROVED** [review]

**Pull request merged**